### PR TITLE
Updates the caching layer to use DB caching for schemas and templates

### PR DESCRIPTION
**Pull request by cybtachyon**
This is a precursor to Patternkit Alpha release.

### CODE REVIEW

File: src/Plugin/PatternLibrary/PatternLibraryFile.php [hunk 1]
```diff
@@ -3,6 +3,7 @@
 namespace Drupal\patternkit\Plugin\PatternLibrary;
 
 use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\patternkit\Entity\PatternInterface;
 use Drupal\patternkit\Pattern;
```
Comment (jordanpagewhite) [inline src/Plugin/PatternLibrary/PatternLibraryFile.php:7]: It seems as though this line `use Drupal\patternkit\Pattern;` was intended to be removed in favor of `use Drupal\patternkit\Entity\PatternInterface;` Is that correct?

File: src/Asset/Library.php [hunk 2]
```diff
@@ -295,196 +305,133 @@ public function buildByExtension($extension_type, $extension): array {
...
 
-    return $libraries;
+    return $this->libraryDefinitions[$extension];
   }
 
   /**
-   * Utility function to get all Patternkit module metadata.
+   * {@inheritDoc}
+   */
+  public function getLibraryAsset($key) {
```
Comment (jordanpagewhite) [inline src/Asset/Library.php:381]: It seems that this method will try to fetch the `atoms/example/ref/src/example/ref` library (i.e. `substr($key, strlen($library_name . '/')) === 'atoms/example/ref/src/example/ref'`  in the current state of this branch, but I believe we want it to fetch `atoms/example/src/example`

File: src/Plugin/PatternLibrary/PatternLibraryFile.php (hunk 1 above)
Comment (cybtachyon) [reply src/Plugin/PatternLibrary/PatternLibraryFile.php:7]: Fixed, thanks!

File: src/Asset/Library.php (hunk 2 above)
Comment (cybtachyon) [reply src/Asset/Library.php:381]: I'm actually not sure what real world scenario we're looking at here - if that's related to an internal library that using /ref/ paths, then this is the wrong place to address it and instead we'd want to add a plugin to handle retrieval.

Or am I reading this wrong?

**Pull request merged**